IPv4 connections to quic-go server fail on FreeBSD

quic-go uses dual-stack sockets so incoming IPv4 connections appear as IPv4-mapped IPv6 connections (see RFC2553 §3.7).  #3881 changed the behaviour of the OOB data associated with the connection (`quic.sconn`) in this case (see the change to `OOB()` in [sys_conn_oob.go](https://github.com/quic-go/quic-go/pull/3881/commits/3b5950a1cea563ef974b8af784a8f8325b843c97#diff-18d15af7517a1da90325d39231f5dba3c53c9be1a3770e256cddddb4d90a670f)):  Previously, an IPv4-mapped address was treated as IPv4 and would result in a `struct in_pktinfo` being attached (this doesn't exist on FreeBSD so the OOB data is empty).  The current behaviour is that an IPv4-mapped address is treated as IPv6 and results in a `struct in6_pktinfo` being attached.  FreeBSD doesn't support `struct in6_pktinfo` for IPv4 addresses and the `sendmsg` call that responds to the incoming connection will return EINVAL.

(Note that FreeBSD recommends that separate IPv4 and IPv6 sockets should be used rather than dual-stack sockets for security reasons - see the *Interaction between IPv4/v6 sockets* section of [inet6(4)](https://man.freebsd.org/cgi/man.cgi?query=inet6)).

Perhaps this information will be useful
1. Change const addr = "localhost:4242" -> const addr = ":4242"  in example/echo/echo.go 
2. Add const addr_dst = "localhost:4242" (addr_dst for connect from client)
3.Change  conn, err := quic.DialAddr(context.Background(), addr, tlsConf, nil) -> conn, err := quic.DialAddr(context.Background(), addr_dst, tlsConf, nil)
4. Run go run example/echo/echo.go

% go run example/echo/echo.go
panic: timeout: no recent network activity

goroutine 1 [running]:
main.main()
        /home/gamer/3/quic/example/echo/echo.go:31 +0x45
exit status 2

 # netstat -na | grep udp
udp46      0      0 *.4242                 *.*

# uname -a
FreeBSD bot 13.1-RELEASE-p6 FreeBSD 13.1-RELEASE-p6 GENERIC amd64

--- a/sys_conn_helper_freebsd.go
+++ b/sys_conn_helper_freebsd.go
@@ -14,7 +14,7 @@ const (
 	ipv4PKTINFO  = 0x7
 )
 
-const ecnIPv4DataLen = 4
+const ecnIPv4DataLen = 1
 
 const batchSize = 8
 

